R/compute_centiloid.R

```diff
@@ -0,0 +1,181 @@
+#' and `custom_intercept` instead. If `custom_slope` and `custom_intercept` are specified,
+#' `tracer`, `pipeline` and `ref_region` are ignored.
+#'
+#' If a matching combination of tracer, pipeline and reference region is not specified and
+#' `custom_slope` and `custom_intercept` are not specified, a warning is issued and
+#' `NA_real_` is returned.
+#'
+#' @return A numeric centiloid value
+#'
+#' @keywords com_bds_findings
+#' @family com_bds_findings
+#'
+#' @export
+#'
+#' @examples
+#' # Using standard parameters
+#' compute_centiloid(
+#'   tracer = "18F-Florbetapir",
+#'   pipeline = "AVID FBP SUVR PIPELINE",
+#'   ref_region = "Whole Cerebellum",
+#'   suvr = 1.25
+#' )
+#'
+#' # Using custom parameters
+#' compute_centiloid(
+#'   tracer = "MyTracer",
+#'   pipeline = "MyPipeline",
+#'   ref_region = "MyRegion",
+#'   suvr = 1.25,
+#'   custom_slope = 193,
+#'   custom_intercept = -187
+#' )
+compute_centiloid <- function(
+    tracer,
+    pipeline,
+    ref_region,
+    suvr,
+    custom_slope = NULL,
+    custom_intercept = NULL) {
+  valid_combinations <- tibble::tribble(
+    ~tracer,              ~pipeline,                     ~ref_region,          ~slope,   ~intercept,
+    "18F-Florbetapir",    "AVID FBP SUVR PIPELINE",      "Whole Cerebellum",   183,      -177,
+    "18F-Florbetapir",    "BERKELEY FBP SUVR PIPELINE",  "Whole Cerebellum",   188.22,   -189.16,
+    "18F-Florbetaben",    "BERKELEY FBB SUVR PIPELINE",  "Whole Cerebellum",   157.15,   -151.87
+  )
+
+  # Check custom_slope and custom_intercept
+  has_custom_slope <- !is.null(custom_slope)
+  has_custom_intercept <- !is.null(custom_intercept)
+
+  if (has_custom_slope != has_custom_intercept) {
+    cli::cli_abort("Both {.code custom_slope} and {.code custom_intercept}
+                   be specified together")
+  }
+
+  use_custom_params <- has_custom_slope && has_custom_intercept
+
+  if (use_custom_params) {
+    assert_numeric_vector(custom_slope, length = 1)
+    assert_numeric_vector(custom_intercept, length = 1)
+    assert_numeric_vector(suvr, length = 1)
+
+    # Use custom parameters
+    slope <- custom_slope
+    intercept <- custom_intercept
+  } else {
+    assert_character_scalar(tracer, values = c("18F-Florbetapir", "18F-Florbetaben"))
+    assert_character_scalar(pipeline, values = c(
+      "AVID FBP SUVR PIPELINE",
+      "BERKELEY FBP SUVR PIPELINE",
+      "BERKELEY FBB SUVR PIPELINE"
+    ))
+    assert_character_scalar(ref_region, values = c(
+      "Whole Cerebellum",
+      "Composite Reference Region"
+    ))
+
+    # Check if the combination exists in valid_combinations
+    check <- valid_combinations %>%
+      dplyr::filter(
+        tracer == !!tracer,
+        pipeline == !!pipeline,
+        ref_region == !!ref_region
+      )
+
+    if (nrow(check) == 0) {
+      cli::cli_warn(
+        c(
+          "No standard conversion formula available for:",
+          "i" = "tracer = {.val {tracer}}",
```

Comment:
Errors look great now! Minor note: I wonder if these would look better as bullets (*)rather than i's? Leaving up to you though, feel free to take or leave